### PR TITLE
Only show editing UI for aliases/related_groups for users /w power

### PR DESCRIPTION
**Pull request by lukebarnard1**

### CODE REVIEW

File: src/components/views/elements/EditableItemList.js [hunk 1]
```diff
@@ -84,7 +84,9 @@ module.exports = React.createClass({
         onNewItemChanged: PropTypes.func,
         onItemAdded: PropTypes.func,
         onItemEdited: PropTypes.func,
-        onItemRemoved: PropTypes. func,
```
Comment (lukebarnard1) [inline src/components/views/elements/EditableItemList.js]: No idea how this is even possible but empirically it's not a syntax error. Babel might be dealing with it?

File: src/components/views/room_settings/AliasSettings.js [hunk 2]
```diff
@@ -262,6 +262,7 @@ module.exports = React.createClass({
                     items={this.state.domainToAliases[localDomain] || []}
                     newItem={this.state.newAlias}
                     onNewItemChanged={this.onNewAliasChanged}
+                    canEdit={this.props.canSetAliases}
```
Comment (turt2live) [inline src/components/views/room_settings/AliasSettings.js]: This seems like it ends up preventing you from adding aliases local to your homeserver. The flag given to the class is for canonical aliases, which should certainly be restricted. However arbitrary aliases are different and bypass power levels: https://github.com/matrix-org/synapse/issues/2402

Comment (lukebarnard1) [reply src/components/views/room_settings/AliasSettings.js]: I'm not convinced this is relevant here. `canSetAliases` is always `true` apparently, so this is more for clarity. `canSetCanonicalAlias` is for canonical aliases.

File: src/components/views/elements/EditableItemList.js (hunk 1 above)
Comment (dbkr) [reply src/components/views/elements/EditableItemList.js]: heh, I can well believe that the dot operator accepts whitespace around it

File: src/components/views/room_settings/AliasSettings.js (hunk 2 above)
Comment (dbkr) [reply src/components/views/room_settings/AliasSettings.js]: Yeah, the current situation is that we hardcode `canSetAliases` to `true` but then also never actually read it anywhere in `AliasSettings`, right? We may as well either wire the thing up if it looks trivial enough to do so (which it looks like it is) or remove it.

File: src/components/views/room_settings/RelatedGroupSettings.js [hunk 3]
```diff
@@ -110,6 +110,7 @@ module.exports = React.createClass({
                 items={this.state.newGroupsList}
                 className={"mx_RelatedGroupSettings"}
                 newItem={this.state.newGroupId}
+                canEdit={this.props.canSetRelatedRooms}
```
Comment (dbkr) [inline src/components/views/room_settings/RelatedGroupSettings.js]: I probably missed this in an earlier PR. but should this be `canSetRelatedGroups`?

Comment (lukebarnard1) [reply src/components/views/room_settings/RelatedGroupSettings.js]: yep!

**Pull request merged**